### PR TITLE
 [Config] Remove dependency to API in non-API code

### PR DESCRIPTION
**Pull request by odesenfans**
Removed all dependencies to the aiohttp app object in non-API code.
This object was used all over the place to retrieve the config
object. Instead, we now use a global variable to store the config
object itself.

### CODE REVIEW

**Comment by odesenfans** [comment]
Depends on #229.

File: src/aleph/config.py [hunk 1]
```diff
@@ -83,3 +85,10 @@ def get_defaults():
             "traces_sample_rate": None,
         },
     }
+
+
+app_config = Config(schema=get_defaults())
```
Comment (hoh) [inline src/aleph/config.py:90]: I see multiple uses of `aleph.config.app_config`. Why not using `aleph.config.get_config()` everywhere and marking this variable as private with a `_` prefix ?

Comment (hoh) [reply src/aleph/config.py:90]: Is this variable useful versus using `@lru_cache` on `def get_config()` ?

File: src/aleph/storage.py [hunk 2]
```diff
@@ -145,7 +146,8 @@ async def verify_content_hash(
     Checks that the hash of a content we fetched from the network matches the expected hash.
     :return: True if the hashes match, False otherwise.
     """
-    ipfs_enabled = app["config"].ipfs.enabled.value
+    config = get_config()
+    ipfs_enabled = config.ipfs.enabled.value
```
Comment (hoh) [inline src/aleph/storage.py:150]: Why not `get_config().ipfs.enabled.value` ?

**Review by hoh — COMMENTED** [review]
Great fix.
A few comments I'd like to discuss.

File: src/aleph/config.py (hunk 1 above)
Comment (odesenfans) [reply src/aleph/config.py:90]: It conveys the intent better IMO. Using LRU cache when there's only one return value seems a bit counterintuitive.

File: src/aleph/storage.py (hunk 2 above)
Comment (odesenfans) [reply src/aleph/storage.py:150]: While coding, my train of thought was 1) get the config and 2) use it where necessary (there are several uses of the config object in most cases). Your suggestion could be useful once we have a dynamic configuration system, reloadable at runtime. Until we have that, I'd keep the local variable.

File: src/aleph/config.py (hunk 1 above)
Comment (hoh) [reply src/aleph/config.py:90]: What about the first comment ?

Comment (odesenfans) [reply src/aleph/config.py:90]: I think the usage in job_utils.py is legit as we need to initialize the global variable. The other one in `get_ipfs_api` looks like an oversight from my part.

**Pull request merged**